lms/djangoapps/course_api/api.py

```diff
@@ -79,6 +83,24 @@ def course_detail(request, username, course_key):
         course_key,
     )
     overview.effective_user = user
+    if user.is_anonymous:
+        overview.certificate_available_date = None
+        return overview
+    current_status = certificate_status_for_student(user, course_key)
+    if settings.FEATURES.get("ENABLE_V2_CERT_DISPLAY_SETTINGS"):
+        display_behavior_is_valid = (
+            overview.certificates_display_behavior == CertificatesDisplayBehaviors.END_WITH_DATE
+        )
+    else:
+        display_behavior_is_valid = True
```

Comment:
I think we should convert this into a common function instead of copying the code itself. The function can then be used in different places like the one I mentioned above in the reference in my last comment. We'll also have to refactor the existing usages of this code, I expect there would not be so many places that will need refactoring.

Although, I agree with @pdpinch that we should not extend the scope of the PR this is something that we should do to avoid unintended problems in the future.